matter_server/server/device_controller.py

```diff
@@ -696,41 +693,70 @@ async def ping_node(self, node_id: int) -> NodePingResult:
         async def _do_ping(ip_address: str) -> None:
             """Ping IP and add to result."""
             timeout = 10 if battery_powered else 2
-            result[ip_address] = await ping_ip(ip_address, timeout)
-
-        # The network interfaces attribute contains a list of network interfaces.
-        # For regular nodes this is just a single interface but we iterate them all anyway.
-        # Create a list of tasks so we can do multiple pings simultanuous.
-        # NOTE: Upgrade this to a TaskGroup once we bump our minimal python version.
-        attr_data = cast(list[dict[str, Any]], node.attributes.get(attr_path))
-        tasks: list[Awaitable] = []
-        for network_interface_data in attr_data:
-            network_interface: Clusters.GeneralDiagnostics.Structs.NetworkInterface = (
-                parse_value(
-                    "network_interface",
-                    network_interface_data,
-                    Clusters.GeneralDiagnostics.Structs.NetworkInterface,
-                )
-            )
-            # ignore invalid/non-operational interfaces
-            if not network_interface.isOperational:
-                continue
-            if network_interface.type in (
-                Clusters.GeneralDiagnostics.Enums.InterfaceTypeEnum.kUnspecified,
-                Clusters.GeneralDiagnostics.Enums.InterfaceTypeEnum.kUnknownEnumValue,
-            ):
-                continue
+            clean_ip = ip_address.split("%")[0]
```

Comment:
The scope ID (followed by the % sign) is important for link-local `fe80` addresses. Depending on the routing table, pinging a fe80 address might work or fail without scope ID.

Since scope ID remains valid on server side, and we get node addresses server side from a locally run mDNS, we should not remove it here but pass it to the ping command.

Reply:
that is exactly what we do. this clean ip is just there for logging

Reply:
Right, sorry I misread that :see_no_evil: 

Hm, but maybe we should also show the scope ID to the user? :thinking: 

We could also go for a compromise and log with scope id, but strip the scope ID from the result. What do you think?

Reply:
I think showing the scope to the user would make them think something is wrong. Let's go with logging it.

Reply:
adjusted the logging to include the interface (in a human readable way)